src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -315,82 +325,82 @@ void EnSw_Destroy(Actor* thisx, PlayState* play) {
             this->collider.base.acFlags &= ~AC_HIT;
-            this->unk_392 = 0x10;
-            Actor_SetColorFilter(&this->actor, COLORFILTER_COLORFLAG_RED, 200, COLORFILTER_BUFFLAG_OPA, this->unk_392);
+            this->painTimer1 = 0x10;
+            Actor_SetColorFilter(&this->actor, COLORFILTER_COLORFLAG_RED, 200, COLORFILTER_BUFFLAG_OPA, this->painTimer1);
             if (Actor_ApplyDamage(&this->actor) != 0) {
                 Actor_PlaySfx(&this->actor, NA_SE_EN_STALTU_DAMAGE);
                 return true;
             }
             Enemy_StartFinishingBlow(play, &this->actor);
-            if (((this->actor.params & 0xE000) >> 0xD) != 0) {
+            if (ENSW_GET_GOLDTYPE(this->actor.params) != SW_NORMALTYPE) {
                 this->skelAnime.playSpeed = 8.0f;
                 if ((play->state.frames & 1) == 0) {
-                    this->unk_420 = 0.1f;
+                    this->rotateMag = 0.1f;
                 } else {
-                    this->unk_420 = -0.1f;
+                    this->rotateMag = -0.1f;
                 }
-                this->unk_394 = 0xA;
-                this->unk_38A = 1;
-                this->unk_420 *= 4.0f;
-                this->actionFunc = func_80B0D878;
+                this->deathFlames = 10;
+                this->animationSpeed = 1;
+                this->rotateMag *= 4.0f;
+                this->actionFunc = EnSw_DieGold;
             } else {
                 this->actor.shape.shadowDraw = ActorShadow_DrawCircle;
                 this->actor.shape.shadowAlpha = 0xFF;
-                this->unk_38A = 2;
+                this->animationSpeed = 2;
                 this->actor.shape.shadowScale = 16.0f;
                 this->actor.gravity = -1.0f;
                 this->actor.flags &= ~ACTOR_FLAG_0;
-                this->actionFunc = func_80B0DB00;
+                this->actionFunc = EnSw_FallNormal;
             }
 
             Actor_PlaySfx(&this->actor, NA_SE_EN_STALWALL_DEAD);
             return true;
         }
     }
 
-    if ((this->unk_390 == 0) && (this->collider.base.atFlags & AT_HIT)) {
-        this->unk_390 = 30;
+    if ((this->painTimer0 == 0) && (this->collider.base.atFlags & AT_HIT)) {
+        this->painTimer0 = 30;
     }
 
     return false;
 }
 
-void func_80B0CBE8(EnSw* this, PlayState* play) {
-    if ((((this->actor.params & 0xE000) >> 0xD) > 0) && (this->actionFunc != func_80B0D590)) {
-        if (this->unk_392 != 0) {
-            this->unk_392--;
+void EnSw_SetCollider(EnSw* this, PlayState* play) {
+    if ((ENSW_GET_GOLDTYPE(this->actor.params) > SW_NORMALTYPE) && (this->actionFunc != EnSw_Crawl)) {
+        if (this->painTimer1 != 0) {
+            this->painTimer1--;
         }
     } else {
-        if ((DECR(this->unk_390) == 0) && (this->actor.colChkInfo.health != 0)) {
+        if ((DECR(this->painTimer0) == 0) && (this->actor.colChkInfo.health != 0)) {
             CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
```

Comment:
`CollisionCheck_SetAT` enables the collider attacking, i.e. damaging Link

so this is more of a delay between damaging Link (on top of Link's own invincibility frames I guess), than related to the skulltula taking damage